### ISSUE TITLE
shellcheck fixer does not work

### ISSUE THREAD

**Issue by johnmaguire** [dense-analysis/ale#4641]
## Information

**VIM version**

```
VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Sep 02 2023 19:52:05)
macOS version - arm64
```

Operating System: MacOS Sonoma 14.0

## What went wrong

ALE isn't running shellcheck when I save shell files.

```
let g:ale_fixers = {
\   'astro': ['prettier'],
\   'javascript': ['prettier'],
\   'javascriptreact': ['prettier'],
\   'css': ['prettier'],
\   'sh': ['shellcheck'],
\}
let g:ale_fix_on_save = 1
```

## Reproducing the bug

1. I opened a file called `main.sh` which was detected by vim as `sh`.
2. I saved the file.
3. I got the following notice:
```
There is no fixer named `shellcheck`. Check :ALEFixSuggest
```

### :ALEInfo
<details>
  <summary>Expand</summary>

 Current Filetype: sh
Available Linters: ['bashate', 'cspell', 'language_server', 'shell', 'shellcheck']
  Enabled Linters: ['bashate', 'cspell', 'language_server', 'shell', 'shellcheck']
  Ignored Linters: []
 Suggested Fixers:
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'shfmt' - Fix sh files with shfmt.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
 
 Linter Variables:
let g:ale_sh_bashate_executable = 'bashate'
let g:ale_sh_bashate_options = ''
let g:ale_sh_language_server_executable = 'bash-language-server'
let g:ale_sh_language_server_use_global = 0
let g:ale_sh_shell_default_shell = 'zsh'
let g:ale_sh_shellcheck_change_directory = 1
let g:ale_sh_shellcheck_dialect = 'auto'
let g:ale_sh_shellcheck_exclusions = ''
let g:ale_sh_shellcheck_executable = 'shellcheck'
let g:ale_sh_shellcheck_options = ''
 
 Global Variables:
let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_disable_lsp = 'auto'
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 1
let g:ale_fixers = {'astro': ['prettier'], 'javascriptreact': ['prettier'], 'sh': ['shellcheck'], 'javascript': ['prettier'], 'css': ['prettier']}
let g:ale_history_enabled = 1
let g:ale_info_default_mode = 'preview'
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {}
let g:ale_linters = {}
let g:ale_linters_explicit = 0
let g:ale_linters_ignore = {}
let g:ale_list_vertical = 0
let g:ale_list_window_size = 10
let g:ale_loclist_msg_format = '%code: %%s'
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 0
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_root = {}
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = 'E'
let g:ale_sign_info = 'I'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = 'E'
let g:ale_sign_style_warning = 'W'
let g:ale_sign_warning = 'W'
let g:ale_sign_highlight_linenrs = 0
let g:ale_type_map = {}
let g:ale_use_neovim_diagnostics_api = 0
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 'all'
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
 
  Command History:

(executable check - failure) bashate
(executable check - failure) cspell
(executable check - failure) bash-language-server
(executable check - success) sh
(finished - exit code 0) ['/bin/zsh', '-c', 'sh -n ''/var/folders/2t/rbxbv8612sq9_rdrxhtw8rzc0000gn/T/vRdrUz2/3/main.sh''']

<<<NO OUTPUT RETURNED>>>

(executable check - success) shellcheck
(finished - exit code 0) ['/bin/zsh', '-c', 'cd ''/Users/jmaguire/src/dnclient/docker'' && ''shellcheck'' --version']

<<<OUTPUT STARTS>>>
ShellCheck - shell script analysis tool
version: 0.9.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
<<<OUTPUT ENDS>>>

(finished - exit code 0) ['/bin/zsh', '-c', 'cd ''/Users/jmaguire/src/dnclient/docker'' && ''shellcheck'' -s sh -x -f gcc - < ''/var/folders/2t/rbxbv8612sq9_rdrxhtw8rzc0000gn/T/vRdrUz2/8/main.sh''']

<<<NO OUTPUT RETURNED>>>

(executable check - failure) bashate
(executable check - failure) cspell
(executable check - failure) bash-language-server
(finished - exit code 0) ['/bin/zsh', '-c', 'sh -n ''/var/folders/2t/rbxbv8612sq9_rdrxhtw8rzc0000gn/T/vRdrUz2/10/main.sh''']

<<<NO OUTPUT RETURNED>>>

(finished - exit code 0) ['/bin/zsh', '-c', 'cd ''/Users/jmaguire/src/dnclient/docker'' && ''shellcheck'' -s sh -x -f gcc - < ''/var/folders/2t/rbxbv8612sq9_rdrxhtw8rzc0000gn/T/vRdrUz2/11/main.sh''']

<<<NO OUTPUT RETURNED>>>

(executable check - failure) bashate
(executable check - failure) cspell
(executable check - failure) bash-language-server
(finished - exit code 0) ['/bin/zsh', '-c', 'sh -n ''/var/folders/2t/rbxbv8612sq9_rdrxhtw8rzc0000gn/T/vRdrUz2/16/main.sh''']

<<<NO OUTPUT RETURNED>>>

(finished - exit code 0) ['/bin/zsh', '-c', 'cd ''/Users/jmaguire/src/dnclient/docker'' && ''shellcheck'' -s sh -x -f gcc - < ''/var/folders/2t/rbxbv8612sq9_rdrxhtw8rzc0000gn/T/vRdrUz2/17/main.sh''']

<<<NO OUTPUT RETURNED>>>

</details>

**Comment by Jorengarenar** [comment]
Well, rather than "not working" the fixer isn't even there. Therefore he message is to be expected, thus it's not really a bug.

The issue probably ought to be feature request; I'd recommend closing and opening a new issue (or converting this one).

**Comment by johnmaguire** [comment]
Hi @Jorengarenar - thanks for the response. Can you clarify what you mean by the "fixer isn't even there"? 

For context, shellcheck is installed on my machine and I believe this configuration was working for years before I began encountering this issue.

**Comment by Jorengarenar** [comment]
Are you sure it was working?  
ShellCheck definitely is supported as linter, but I don't see any reference of it being implemented also as a fixer.
